it/src/main/java/com/google/cloud/teleport/it/datastream/DatastreamResourceManager.java

```diff
@@ -0,0 +1,392 @@
+                    mysqlProfileBuilder.setHostname(source.hostname()).setUsername(source.username()).setPassword(source.password()).setPort(source.port());
+                    connectionProfileBuilder.setMysqlProfile(mysqlProfileBuilder);
+                    break;
+                case ORACLE:
+                    OracleProfile.Builder oracleProfileBuilder = OracleProfile.newBuilder();
+                    oracleProfileBuilder.setHostname(source.hostname()).setUsername(source.username()).setPassword(source.password()).setPort(source.port());
+                    connectionProfileBuilder.setOracleProfile(oracleProfileBuilder);
+                    break;
+                case POSTGRESQL:
+                    PostgresqlProfile.Builder postgresqlProfileBuilder = PostgresqlProfile.newBuilder();
+                    postgresqlProfileBuilder.setHostname(source.hostname()).setUsername(source.username()).setPassword(source.password())
+                            .setPort(source.port()).setDatabase(((PostgresSource) source).getDatabase());
+                    connectionProfileBuilder.setPostgresqlProfile(postgresqlProfileBuilder);
+                    break;
+            }
+
+            ConnectionProfile connectionProfile = connectionProfileBuilder
+                    .setDisplayName(connectionProfileId)
+                    .setStaticServiceIpConnectivity(StaticServiceIpConnectivity.getDefaultInstance())
+                    .build();
+            CreateConnectionProfileRequest request = CreateConnectionProfileRequest.newBuilder()
+                    .setParent(LocationName.of(projectId, location).toString())
+                    .setConnectionProfile(connectionProfile)
+                    .setConnectionProfileId(connectionProfileId)
+                    .build();
+            ConnectionProfile reference = datastreamClient.createConnectionProfileAsync(request).get();
+            createdConnectionProfileIds.add(connectionProfileId);
+
+            LOG.info("Successfully created JDBC Source Connection Profile {} in project {}.", connectionProfileId, projectId);
+            return reference;
+        } catch (ExecutionException | InterruptedException e) {
+            throw new DatastreamResourceManagerException("Failed to create JDBC source connection profile. ", e);
+        }
+    }
+
+    public synchronized SourceConfig buildSourceConfig(String sourceConnectionProfileId, JDBCSource source) {
+        checkArgument(source.getAllowedTables() == null && source.getAllowedTables().isEmpty(),
+                "JDBC source allowed tables can not be null or empty");
+        Map<String, List<String>> allowedTables = source.getAllowedTables();
+        SourceConfig.Builder sourceConfigBuilder = SourceConfig.newBuilder()
+                .setSourceConnectionProfile(ConnectionProfileName
+                        .format(projectId, location, sourceConnectionProfileId));
+
+        switch (source.type()) {
```

Comment:
ok this sounds good. 

Removing `allowedTables` from `JDBCSource` means i'll have to pass it as an argument in `config()`. Allowed tables is not optional and is needed to configure the source. I can either have `allowedTables` as an attribute in all 3 subclasses or argument in config(). the latter seems easier. Lmk if you recommend otherwise

![image](https://github.com/GoogleCloudPlatform/DataflowTemplates/assets/42254790/b02116cc-4ef2-4707-87bf-a4a1cc0bbdbb)

Reply:
My comments are slightly contradictory...

I think the `JDBCSource.Builder` could retain the `setAllowedTables()` method, but it would be abstract and take in a type `T` that could be defined in the subclasses so that they all define what exactly the `allowedTables` are. Since some are schemas and some are table names, I think that would make it clearer.